src/pages/pu-trx/[functionId]/PurchaseTransactionForm.js

```diff
@@ -0,0 +1,1662 @@
+                  isPosted
+                }
+                required={
+                  !formik?.values?.header.dtId ||
+                  (formik?.values?.header.dtId && formik?.values?.header.commitItems == true)
+                }
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('header.siteId', newValue ? newValue.recordId : null)
+                  formik.setFieldValue('header.siteRef', newValue ? newValue.reference : null)
+                  formik.setFieldValue('header.siteName', newValue ? newValue.name : null)
+                }}
+                error={formik.touched?.header?.siteId && Boolean(formik.errors?.header?.siteId)}
+              />
+            </Grid>
+            <Grid item xs={2.4}>
+              <ResourceComboBox
+                endpointId={SystemRepository.Currency.qry}
+                name='currencyId'
+                label={labels.currency}
+                valueField='recordId'
+                displayField={['reference', 'name']}
+                readOnly={isPosted}
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' }
+                ]}
+                required
+                values={formik.values.header}
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('header.currencyId', newValue?.recordId || null)
+                }}
+                error={formik.touched?.header?.currencyId && Boolean(formik.errors?.header?.currencyId)}
+              />
+            </Grid>
+            <Grid item xs={2.4}></Grid>
+
+            <Grid item xs={4.8}>
+              <ResourceLookup
+                endpointId={PurchaseRepository.Vendor.snapshot}
+                name='vendorId'
+                label={labels.vendor}
+                valueField='reference'
+                displayField='name'
+                valueShow='vendorRef'
+                secondValueShow='vendorName'
+                formObject={formik.values.header}
+                form={formik}
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' },
+                  { key: 'flName', value: 'FL Name' }
+                ]}
+                onChange={(event, newValue) => {
+                  fillVendorData(newValue)
+                }}
+                secondFieldName={'header.vendorName'}
+                onSecondValueChange={(name, value) => {
+                  formik.setFieldValue('header.vendorName', value)
+                }}
+                errorCheck={'header.vendorId'}
+                maxAccess={maxAccess}
+                required
+                readOnly={isPosted}
+                displayFieldWidth={3}
+                editMode={editMode}
+              />
```

Comment:
![image](https://github.com/user-attachments/assets/057c58bf-1306-4a58-8237-8844748cbe7f)
just make sure there is neef for flName